packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -392,7 +392,7 @@ class HotRunner extends ResidentRunner {
     // We are now running from source.
     _runningFromSnapshot = false;
     final String launchPath = debuggingOptions.buildInfo.previewDart2
-        ? dillOutputPath ?? mainPath + '.dill'
+        ? mainPath + '.dill'
```

Comment:
Why did we remove support for `dillOutputPath` here?  /cc @keertip

Reply:
The reason was that in this particular case absolute file path [dillOutputPath] is wrong, because it has to be relative to the device's devfs. That is what [mainPath + '.dill'] is.